toolchain/base/value_store.h

```diff
@@ -33,36 +33,60 @@ class ValueStoreNotPrintable {};
 }  // namespace Internal
 
 struct IdTag {
-  IdTag()
-      : id_tag_(0),
-        initial_reserved_ids_(std::numeric_limits<int32_t>::max()) {}
+  IdTag() = default;
+
   explicit IdTag(int32_t id_index, int32_t initial_reserved_ids)
       : initial_reserved_ids_(initial_reserved_ids) {
     // Shift down by 1 to get out of the high bit to avoid using any negative
-    // ids, since they have special uses.
-    // Shift down by another 1 to free up the second highest bit for a marker to
-    // indicate whether the index is tagged (& needs to be untagged) or not.
-    // Add one to the index so it's not zero-based, to make it a bit less likely
-    // this doesn't collide with anything else (though with the
-    // second-highest-bit-tagging this might not be needed).
+    // ids, since they have special uses. Shift down by another 1 to free up the
+    // second highest bit for a marker to indicate whether the index is tagged
+    // (& needs to be untagged) or not. Add one to the index so it's not
+    // zero-based, to make it a bit less likely this doesn't collide with
+    // anything else (though with the second-highest-bit-tagging this might not
+    // be needed).
     id_tag_ = llvm::reverseBits((((id_index + 1) << 1) | 1) << 1);
```

Comment:
If you're editing constructors, perhaps this should be done via initializer list instead of assignment?

Reply:
Done